pytket/phir/phirgen_parallel.py

```diff
@@ -353,7 +353,7 @@ def genphir_parallel(
         )
     adjust_phir_transport_time(ops, machine)
 
-    decls = get_decls(qbits, cbits)
+    decls = get_decls(circuit.q_registers, circuit.c_registers)
```

Comment:
A possible "gotcha" here is that if a pytket circuit contains bits that do not form part of a full register -- for example, if it contains a bit `c[1]` but not `c[0]` -- then these bits are _ignored_ by `circuit.c_registers` (similarly for qubits). I am not sure how phirgen handles this case currently (maybe an error, maybe add missing bits?) -- but if such a circuit were passed to this function it would presumably generate invalid PHIR.

Reply:
Thanks @cqc-alec. My guess without working with an example is that as long as pytket commands applying operations are referring to the right bits, we will generate correct PHIR commands for them. The only chance of error here, from what I can gather from your comment, is that the declaration command may mismatch the dimension of a register.

Would you be able to provide an example to help make this change more robust?

Reply:
Sure, how about this:
```python
from pytket.circuit import Circuit, Qubit, Bit
from pytket.phir.api import pytket_to_phir

circ = Circuit()
circ.add_qubit(Qubit(1))
circ.add_bit(Bit(1))
circ.H(1)
circ.Measure(1, 1)
phir = pytket_to_phir(circ)
print(phir)
```

Currently this produces the following:
```
{"format": "PHIR/JSON", "version": "0.1.0", "metadata": {"source": "pytket-phir v0.8.1"}, "ops": [{"data": "qvar_define", "data_type": "qubits", "variable": "q", "size": 1}, {"data": "cvar_define", "data_type": "i64", "variable": "c", "size": 1}, {"//": "H q[1];"}, {"qop": "H", "angles": null, "args": [["q", 1]]}, {"//": "Measure q[1] --> c[1];"}, {"qop": "Measure", "returns": [["c", 1]], "args": [["q", 1]]}]}
```
This looks incorrect, since it seems to be indexing `q[1]` in a size-1 register. I am not sure what pecos will do with this, probably error?

With the change in this PR, I suspect it will be differently wrong, since the qubit and bit registers would not be declared at all?

I think it would be OK to reject circuits with incomplete registers.

Reply:
> ```python
> circ.add_qubit(Qubit(1))
> circ.add_bit(Bit(1))
> ```

One would think that these operations should not be allowed by `pytket` as they seem inherently unsafe, and no backend may be able to simulate/run them.

What's the rationale behind supporting them in pytket? In general, I agree we should reject this circuit in either of the cases -- with or without this PR.

Reply:
The fundamental entities in pytket are not registers but `Qubit`s and `Bit`s -- and each of these is identified by the combination of a string and an index. There's nothing inherently wrong with this but it is confusing because most other languages and representations have the register as their fundamental entity, and in pytket this is a secondary concept.

As for the original rationale -- I'm afraid I don't know!

Reply:
Thanks, I will update the PR to reject such programs.

Reply:
OK, equally happy for that to be a separate PR.

Reply:
Let me make another PR then.